info: Server started at undefined:undefined/api/undefined/ {"timestamp":"2024-03-27T14:10:12.430Z"}
error: Could not connect to the database. The `uri` parameter to `openUri()` must be a string, got "undefined". Make sure the first parameter to `mongoose.connect()` or `mongoose.createConnection()` is a string. {"stack":"MongooseError: The `uri` parameter to `openUri()` must be a string, got \"undefined\". Make sure the first parameter to `mongoose.connect()` or `mongoose.createConnection()` is a string.\n    at NativeConnection.Connection.openUri (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\connection.js:694:11)\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:351:10\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\helpers\\promiseOrCallback.js:32:5\n    at new Promise (<anonymous>)\n    at promiseOrCallback (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\helpers\\promiseOrCallback.js:31:10)\n    at Mongoose._promiseOrCallback (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:1149:10)\n    at Mongoose.connect (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:350:20)\n    at Database.<anonymous> (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:20:22)\n    at Generator.next (<anonymous>)\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:8:71\n    at new Promise (<anonymous>)\n    at __awaiter (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:4:12)\n    at Database.initializeDatabase (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:18:57)\n    at App.initializeDatabase (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\index.ts:62:13)\n    at new App (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\index.ts:38:10)\n    at Object.<anonymous> (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\index.ts:90:13)","timestamp":"2024-03-27T14:10:12.442Z"}
info: Server started at http://localhost:4000/api/v1/ {"timestamp":"2024-03-27T14:22:24.742Z"}
error: Could not connect to the database. Invalid connection string {"name":"MongoParseError","stack":"MongoParseError: Invalid connection string\n    at parseConnectionString (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongodb\\lib\\core\\uri_parser.js:585:21)\n    at connect (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongodb\\lib\\operations\\connect.js:283:3)\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongodb\\lib\\mongo_client.js:284:5\n    at maybePromise (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongodb\\lib\\utils.js:692:3)\n    at MongoClient.connect (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongodb\\lib\\mongo_client.js:280:10)\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\connection.js:836:12\n    at new Promise (<anonymous>)\n    at NativeConnection.Connection.openUri (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\connection.js:832:19)\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:351:10\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\helpers\\promiseOrCallback.js:32:5\n    at new Promise (<anonymous>)\n    at promiseOrCallback (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\helpers\\promiseOrCallback.js:31:10)\n    at Mongoose._promiseOrCallback (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:1149:10)\n    at Mongoose.connect (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:350:20)\n    at Database.<anonymous> (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:20:22)\n    at Generator.next (<anonymous>)","timestamp":"2024-03-27T14:22:24.752Z"}
info: Server started at http://localhost:4000/api/v1/ {"timestamp":"2024-03-27T14:23:10.241Z"}
error: Could not connect to the database. Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://docs.atlas.mongodb.com/security-whitelist/ {"reason":{"commonWireVersion":null,"compatibilityError":null,"compatible":true,"heartbeatFrequencyMS":10000,"localThresholdMS":15,"logicalSessionTimeoutMinutes":null,"maxElectionId":null,"maxSetVersion":null,"servers":{},"setName":null,"stale":false,"type":"ReplicaSetNoPrimary"},"stack":"MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://docs.atlas.mongodb.com/security-whitelist/\n    at NativeConnection.Connection.openUri (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\connection.js:847:32)\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:351:10\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\helpers\\promiseOrCallback.js:32:5\n    at new Promise (<anonymous>)\n    at promiseOrCallback (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\helpers\\promiseOrCallback.js:31:10)\n    at Mongoose._promiseOrCallback (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:1149:10)\n    at Mongoose.connect (C:\\Users\\USER\\Documents\\software\\catura-api\\node_modules\\mongoose\\lib\\index.js:350:20)\n    at Database.<anonymous> (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:20:22)\n    at Generator.next (<anonymous>)\n    at C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:8:71\n    at new Promise (<anonymous>)\n    at __awaiter (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:4:12)\n    at Database.initializeDatabase (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\config\\database.ts:18:57)\n    at App.initializeDatabase (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\index.ts:62:13)\n    at new App (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\index.ts:38:10)\n    at Object.<anonymous> (C:\\Users\\USER\\Documents\\software\\catura-api\\src\\index.ts:90:13)","timestamp":"2024-03-27T14:23:43.275Z"}
info: Server started at http://localhost:4000/api/v1/ {"timestamp":"2024-03-27T14:27:27.860Z"}
info: Connected to the database. {"timestamp":"2024-03-27T14:27:30.569Z"}
